### PR TITLE
Run I/O tests in CI #146

### LINKED ISSUE #146: Run I/O tests in CI
Currently, only JUnit tests are run by Travis.
It's nice if Travis can also run I/O tests.

### PR DESCRIPTION
**Pull request by chao1995**
Fixes #146 

Ready for review.

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@chao1995 submitted v1 for review.

* [v1 1/1] [Travis: run system tests](https://github.com/se-edu/addressbook-level2/pull/189/commits/c1bb92140c795e1cd1370fa222ddc78401bfee85)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/d23881802cf0b0960cf2deb64294cc12b1313c0c...c1bb92140c795e1cd1370fa222ddc78401bfee85))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/189/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by yamgent** [comment]
The PR title should be "Run I/O tests in CI #146".

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,11 @@ language: java
 matrix:
   include:
     - jdk: oraclejdk8
-    
+
+script:
+  - cd test && ./runtests.sh && cd .. # system tests
+  - ./gradlew clean test # JUnit tests
```
Comment (damithc) [inline .travis.yml]: We can either use `system tests` and `unit/integration tests` OR `I/O tests` and `JUnit tests`?

Comment (damithc) [inline .travis.yml]: Same goes for the commit message

**Comment by damithc** [comment]
@m133225 can have a look too?
@chao1995 what is the possibility of running tests on AppVeyor as well? I/O tests are particularly vulnerable to OS quirks. Can be a separate PR.

File: .travis.yml (hunk 1 above)
Comment (chao1995) [inline .travis.yml]: > We can either use system tests and unit/integration tests OR I/O tests and JUnit tests

Agree. Didn't realise this before. We also need to update our [testing docs](https://github.com/se-edu/addressbook-level2/blob/master/doc/DeveloperGuide.md#testing) too. It uses system tests together with JUnit tests.

**Comment by chao1995** [comment]
> what is the possibility of running tests on AppVeyor as well?

It should be easy to set up. similar to `runtest.sh`, we have a `runtest.bat`.

**Comment by CanIHasReview** [comment]
# v2

@chao1995 submitted v2 for review.

* [v2 1/1] [Travis: run I/O tests](https://github.com/se-edu/addressbook-level2/pull/189/commits/86caea42c6b432f13fa6084837ac8691d384d142)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/d23881802cf0b0960cf2deb64294cc12b1313c0c...86caea42c6b432f13fa6084837ac8691d384d142)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level2/compare/0e3e41b59423da1146f01cee2ec4d94866959962~1...0e3e41b59423da1146f01cee2ec4d94866959962))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/189/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v3

@chao1995 submitted v3 for review.

* [v3 1/1] [Travis: run I/O tests](https://github.com/se-edu/addressbook-level2/pull/189/commits/a165d88d15a93eff1945030a6fa21067645bdb4b)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/d23881802cf0b0960cf2deb64294cc12b1313c0c...a165d88d15a93eff1945030a6fa21067645bdb4b)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level2/compare/7707e6d30f34d38a22754b60f2fe9f85c89d3545~1...7707e6d30f34d38a22754b60f2fe9f85c89d3545))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/189/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by chao1995** [comment]
# v3

Simplifie `cd` in and out of test folder by utilise shell script's `(` operator, which executes the commands in a subshell.

**Comment by chao1995** [comment]
# v3

Simplify `cd` in and out of test folder by utilise shell script's `(` operator, which executes the commands in a subshell.

**Comment by chao1995** [comment]
# v3

Simplify `cd` in and out of test folder by utilising shell script's `(` operator, which executes the commands in a subshell.

**Comment by CanIHasReview** [comment]
# v4

@chao1995 submitted v4 for review.

* [v4 1/1] [Travis: run I/O tests](https://github.com/se-edu/addressbook-level2/pull/189/commits/b32b5e5533d24adad0eb45c69a579048957c5abf)

([:books: Archive](https://github.com/se-edu/addressbook-level2/compare/816c5037f49e2740cffd3cd93c99e5261f531b20...b32b5e5533d24adad0eb45c69a579048957c5abf)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level2/compare/34f9ed707a7b8c2b27f8c68717900dbce6e75230~1...34f9ed707a7b8c2b27f8c68717900dbce6e75230))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level2.git refs/pr/189/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Pull request merged**